blocks/bricks/recurrent.py

```diff
@@ -198,12 +200,22 @@ def scan_function(*args):
             outputs_info = (list(states_given.values()) +
                             [None] * (len(application.outputs) -
                                       len(application.states)))
+            if config.scan_strict:
+                # Find all shared variables
+                shared_vars = list(Selector(brick).get_params().values())
```

Comment:
Agreed that `strict = kwargs.pop('strict', config.scan_strict)` would be nice. Don't forget to add it to the docstring in that case.